File: 2728-sum-in-a-matrix/sum-in-a-matrix.java
class Solution {
    public int matrixSum(int[][] nums) {
        int row = nums.length;
        int col = nums[0].length;
        int total = 0;

        for (int[] arr : nums) {
            Arrays.sort(arr);
        }

        for (int i = 0; i < col; i++) {
            int mx = 0;
            for (int j = 0; j < row; j++) {
                mx = Math.max(mx, nums[j][i]);
            }
            // cout<<mx<<endl;
            total += mx;
        }
        return total;
    }
}

File: 2728-sum-in-a-matrix/sum-in-a-matrix.cpp
class Solution {
public:
    int matrixSum(vector<vector<int>>& nums) {
        int row = nums.size();
        int col = nums[0].size();

        int total = 0;
        for(int i=0;i<nums.size();i++){
            sort(nums[i].begin(), nums[i].end());
        }

        for(int i=0;i<col;i++){
            int mx = 0;
            for(int j=0;j<row;j++){
                mx = max(mx,nums[j][i]);
            }
            total+=mx;
        }
        return total;

    }
};